docs/proposals/config_data.md

```diff
@@ -0,0 +1,298 @@
+		TypeMeta `json:",inline"`
+		ListMeta `json:"metadata,omitempty"`
+
+		Items []ConfigData `json:"items"`
+	}
+```
+
+A new API resource selector (``ConfigDataSelector``) will be added to select objects and fields from those objects
+for consuming API resources through the downward API.
+
+``Registry`` interface will be added to accompany the ``ConfigData`` resource.  
+
+### Volume Source
+
+A new ``ConfigDataVolumeSource`` type of volume source containing the ``ConfigData`` object will be added to the ``VolumeSource`` struct in the API:
+
+```go
+	type ConfigDataVolumeSource struct {
+		ConfigDataName string `json:"configDataName"`
+	}
+```
+
+**Note:** The update logic used in the downward API volume plug-in will be extracted and re-used in the volume plug-in for ``ConfigData``.
+
+## Creating ConfigData
+The following file ``pod-config-example.json`` contains data intended to be consumed as environment variables.
+```json
+{
+    "apiVersion": "extensions/v1beta1",
+    "metadata": {
+        "name": "pod-config-example",
+    },
+    "kind": "ConfigData",
+    "data": {
+        "server_address": "10.240.13.14:8998",
+        "db_address": "10.240.55.3",
+        "db_port": "3306"
+    }
+}
+```
+
+To persist this ConfigData object to the API server, use the following command:
+```bash
+kubectl create -f pod-config.json
+```
+
+Now, the ``pod-config-example`` ConfigData object is stored in ``etcd``.
+The new object can be accessed, referenced, used and updated through the API server.
+
+## Configuration Workflow
+
+### Environment Variables
+
+The component configuration workflow is given below:
+
+1. ``ConfigData`` object is created through the API server.
+2. New components using the ``ConfigData`` object are created.
+3. ``ConfigData`` object is retrieved using the downward API.
+4. ``Data`` key/value pairs of ``ConfigData`` will be exposed as environment variables (``EnvVarSource``).
+5. API server is watched for modifications to the object.
+6. After a modification, actions 3, 4, and 5 are repeated.
+
+### Mounted Volume
+
+The component configuration workflow is given below:
+
+1. ``ConfigData`` object is created through the API server.
+2. New components using the ``ConfigData`` object are created.
+3. ``ConfigData`` object is retrieved using the downward API.
+4. ``ConfigData`` will be mounted as a new volume (``ConfigDataVolumeSource``) where keys are file names and values are file contents.
+5. API server is watched for modifications to the object.
+6. After a modification, actions 3, 4, and 5 are repeated.
+
+## Examples
+
+#### Consuming ConfigData as mounted volumes
+
+``pod-config-volume`` is intended to be used as a mounted volume containing two files with their respective contents.
+```json
+{
+    "apiVersion": "extensions/v1beta1",
+    "metadata": {
+        "name": "pod-config-volume",
+    },
+    "kind": "ConfigData",
+    "data": {
+        "component_config": "COMPONENT_CONFIG_STRING_CONTENTS",
```

Comment:
I think this would be better as a real example.  How about a small, real example of a config file?